src/js/pages/UserContext.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect, useContext } from 'react';
+import { useAuth0 } from 'js/services/react-auth0-spa';
+
+import Simple from 'js/pages/templates/Simple';
+
+import { getUserWithAccountRepos } from 'js/services/api';
+
+const context = React.createContext(null);
+
+export const useUser = () => useContext(context);
+
+export default ({ children }) => {
+    const { logout, loading, isAuthenticated, getTokenSilently } = useAuth0();
+    const [userState, setUserState] = useState(null);
+
+    useEffect(() => {
+        if (loading) {
+            return;
+        };
+
+        if (!isAuthenticated) {
+            setUserState(null);
+            return;
+        };
+
+        getTokenSilently()
+            .then(getUserWithAccountRepos)
+            .then(setUserState)
+            .catch(err => console.error('Could not get user with repos', err));
+
+    }, [loading, isAuthenticated, getTokenSilently]);
```

Comment:
Adding the `getTokenSilently` function shouldn't be necessary.

```suggestion
    }, [loading, isAuthenticated]);
```

Reply:
It's required by `react-hooks/exhaustive-deps` lint rule.
(more info at https://github.com/facebook/react/issues/14920#issuecomment-471070149 when answering _"My function value is constant"_)

Reply:
Interesting, thanks for pointing to the very specific section of that very long thread.